src/components/ChangePlan/Checkout/CheckoutSummary.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { extractParameter } from '../../../utils';
+import queryString from 'query-string';
+import SafeRedirect from '../../SafeRedirect';
+
+export const CheckoutSummary = ({ location }) => {
```

Comment:
The implementation is a bit complicated for what the component does.What do you think about this:

``` 
export const CheckoutSummary = ({ location }) => {
  const redirect = extractParameter(location, queryString.parse, 'redirect');
  const legacy = extractParameter(location, queryString.parse, 'legacy');

  if(legacy) {
     if(redirect) {
       return <SafeRedirect to={redirect} />
     }
     return <SafeRedirect to="/Campaigns/Draft" />
  }
  
  return ('Normal checkout flow for new plans should go here.');
};
```